source/lib/course-search/update-course-storage.js

```diff
@@ -76,22 +76,12 @@ function storeTermInfo(terms: Array<TermType>) {
 	RNFS.writeFile(TERMS_FILE, json, 'utf8')
 }
 
-async function storeTermCoursesFromServer(path: string): boolean {
+async function storeTermCoursesFromServer(path: string): Promise<boolean> {
 	const url = COURSE_DATA_PAGE + path
-
-	try {
-		const resp: Array<CourseType> = await fetchJson(url)
-		const courseJson = JSON.stringify(resp)
-		const filePath = COURSE_STORAGE_DIR + path
-		RNFS.writeFile(filePath, courseJson, 'utf8')
-			.then(() => {
-				return true
-			})
-			.catch(() => {
-				return false
-			})
-	} catch (error) {
-		console.log(error)
-		return false
-	}
+	const resp: Array<CourseType> = await fetchJson(url)
+	const courseJson = JSON.stringify(resp)
```

Comment:
If you're fetching and then re-stringifying the JSON, I'd use the `fetchText` helper instead – it's also a global, just like fetchJson, but it doesn't parse the result as JSON.

Reply:
I've been trying to use fetchText instead for a good minute now and I'm getting some suuuuper weird results where if I go offline and clear the cached data from the simulator's filesystem, it fetches some of the course data and stores it, but it doesn't get all of it and I don't know 1. how it's fetching the data with no internet connection and 2. why it's fetching some of the data, but not all of it. I'm tempted to just leave it as is instead of spending a bunch of time debugging this edge case...what do you think?